semialign/src/Data/Zip.hs

```diff
@@ -35,7 +35,6 @@ instance (Zip f, Semigroup a) => Semigroup (Zippy f a) where
 
 instance (Repeat f, Monoid a) => Monoid (Zippy f a) where
     mempty                      = Zippy $ repeat mempty
-    mappend (Zippy x) (Zippy y) = Zippy $ zipWith mappend x y
```

Comment:
this change is potentially changing behavior, so no, I won't do these. Until GHC starts to treat them as errors, I won't cleanup these warnings.